Dockerfile

```diff
@@ -14,31 +15,42 @@ ENV BUILD_ARCH=${BUILD_ARCH}
 
 ENV BUILD_TARGET=${BUILD_ARCH}-unknown-linux-gnu
 
-RUN apt-get update \
-    && apt-get install --no-install-recommends -y \
-    curl build-essential git zip cmake \
+RUN yum -y update \
+    && yum -y install centos-release-scl epel-release \
+    # install a modern compiler toolchain
+    && yum -y install cmake3 devtoolset-10 git \
     # below required for sentry-native
-    clang libcurl4-openssl-dev \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/*
+    llvm-toolset-7.0-clang-devel \
+    && yum clean all \
+    && rm -rf /var/cache/yum \
+    && ln -s /usr/bin/cmake3 /usr/bin/cmake
+
+ENV RUSTUP_HOME=/usr/local/rustup \
+    CARGO_HOME=/usr/local/cargo \
+    PATH=/usr/local/cargo/bin:$PATH
+
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --profile minimal
+
+COPY --from=sentry-cli /bin/sentry-cli /bin/sentry-cli
```

Comment:
Moved this from the relay-builder stage to relay-deps. It looks like this was placed there to ensure we have a more recent `sentry-cli`, but since sentry-cli is now more stable, we can also copy it into the deps right away. 

I can look into updating to sentry-cli v2 in a separate step.